util/genrest/testdata/TestConstructServerStreamer.go.baseline

```diff
@@ -0,0 +1,110 @@
+// Catalog_BaseServerStreamer contains the basic accumulation and emit functionality to help handle all server streaming RPCs in the Catalog service.
+type Catalog_BaseServerStreamer struct{
+   responses []string
+   initialization sync.Once
+   marshaler      *protojson.MarshalOptions
+ 
+   grpc.ServerStream
+ }
+
+func (streamer *Catalog_BaseServerStreamer) accumulate(response proto.Message) error {
+   streamer.initialization.Do(streamer.initialize)
+   json, err := streamer.marshaler.Marshal(response)
+   if err != nil {
+     return fmt.Errorf("error json-encoding response: %s", err.Error())
+   }
+   streamer.responses = append(streamer.responses, string(json))
+   return nil
+}
+
+// ListJSON returns a list of all the accumulated responses, in JSON format.
+func (streamer *Catalog_BaseServerStreamer) ListJSON() string {
+   return fmt.Sprintf("{\n%s\n}", strings.Join(streamer.responses, ",\n"))
+}
+
+func (streamer *Catalog_BaseServerStreamer) initialize() {
+   streamer.marshaler = resttools.ToJSON()
+}
```

Comment:
I don't think it matters too much, but why have the accumulate do initialize once/idemontently, instead of creating the streamer via a factory function and returning it with the correct marshaler?

Reply:
No strong reason; we could do it either way.

In some sense it seems cleaner and more concise this way, because otherwise we would have one  factory function for `Echo_ExpandServer` (and other streaming RPCs) , and each of those would either call the factory function for  `Echo_BaseServerStreamer` or explicitly initialize its internals. This way it's more self-contained , and `accumulate()` enforces its prerequisites being satisfied.